shm/include/locators.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "time.hpp"
```

Comment:
instead of include we can use forward declaration
class Time;

Reply:
Probably it will need much more alignment:

Reply:
In the .cpp files you have to include the whole file otherwise it will not work